ci-operator/step-registry/openstack/test/nfv/openstack-test-nfv-commands.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+function check_pod_status() {
+    INTERVAL=60
+    CNT=10
+    while [ $((CNT)) -gt 0 ]; do
+        READY=false
+        while read -r i
+        do
+            pod_name=$(echo "${i}" | awk '{print $1}')
+            pod_phase=$(echo "${i}" | awk '{print $3}')
+            if [[ "${pod_phase}" == "Running" ]]; then
+                READY=true
+            else
+                echo "Waiting for Pod ${pod_name} to be ready"
+                READY=false
+            fi
+        done <<< "$(oc -n "${CNF_NAMESPACE}" get pods "${CNF_POD}" --no-headers)"
+
+        if [[ "${READY}" == "true" ]]; then
+            echo "Pod ${CNF_POD} has successfully been deployed"
+            return 0
+        else
+            sleep "${INTERVAL}"
+            CNT=$((CNT))-1
+        fi
+
+        if [[ $((CNT)) -eq 0 ]]; then
+            echo "Pod ${CNF_POD} did not successfully deploy"
+            oc -n "${CNF_NAMESPACE}" get pods "${CNF_POD}"
+            return 1
+        fi
+    done
+}
+
+CNF_NAMESPACE="example-cnf"
+export OS_CLIENT_CONFIG_FILE="${SHARED_DIR}/clouds.yaml"
+
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+	# shellcheck disable=SC1090
+	source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+CNF_NAMESPACE=$(
+    oc create -f - -o jsonpath='{.metadata.name}' <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ${CNF_NAMESPACE}
+EOF
+)
+echo "Created \"$CNF_NAMESPACE\" Namespace"
+
+if ! openstack network show "${OPENSTACK_PERFORMANCE_NETWORK}" >/dev/null 2>&1; then
+    echo "Network ${OPENSTACK_PERFORMANCE_NETWORK} doesn't exist"
+    exit 1
+fi
+
+SUBNET_ID=$(openstack network show "${OPENSTACK_PERFORMANCE_NETWORK}" -f json -c subnets | jq '.subnets[0]' | sed 's/"//g')
+SUBNET_TAG=$(openstack subnet show "${SUBNET_ID}" -f json -c tags)
+WHEREABOUTS_RANGE=$(echo "${SUBNET_TAG}" | grep 'no_dhcp_range')
+if [ -z "${WHEREABOUTS_RANGE}" ]; then
+    echo "Subnet ${SUBNET_ID} doesn't have a no_dhcp_range tag"
+    exit 1
+fi
+# shellcheck disable=SC2001
+WHEREABOUTS_RANGE=$(echo "${WHEREABOUTS_RANGE#*=}" | sed 's/"//g')
+cat <<EOF > "${SHARED_DIR}/additionalnetwork.yaml"
+spec:
+  additionalNetworks:
+  - name: ${OPENSTACK_PERFORMANCE_NETWORK}
+    namespace: ${CNF_NAMESPACE}
+    rawCNIConfig: '{ "cniVersion": "0.3.1", "name": "${OPENSTACK_PERFORMANCE_NETWORK}", "type": "host-device","pciBusId": "0000:00:04.0", "ipam": { "type": "whereabouts", "range": "${WHEREABOUTS_RANGE}"}}'
```

Comment:
according to https://docs.openshift.com/container-platform/4.9/networking/multiple_networks/configuring-host-device.html#nw-multus-host-device-object_configuring-host-device   
"Specify your network device by setting only one of the following parameters: device, hwaddr, kernelpath, or pciBusID." are any of the other easier to predict than the pciBusId

Reply:
I think for now, we can hardcode `0000:00:04.0` as all the deployments that I saw where the worker has a secondary network, the bus was `0000:00:04.0`.
We'll investigate how to find out the PCI Bus ID indeed, so it'll be a bit smarter than it is today for this initial implementation.

Reply:
We could re-use some of the snippets in https://github.com/rh-nfv-int/shift-on-stack-vhostuser/blob/master/roles/sos-vhostuser/files/vhostuser that helps to figure out the BusId; but this would require shell access on the worker. I guess we could do it with `oc rsh (...)` at some point.